packages/main/src/RPA/Robocorp/WorkItems.py

```diff
@@ -1255,3 +1271,32 @@ def remove_work_item_files(self, pattern, missing_ok=True):
 
         logging.info("Removed %d file(s)", len(names))
         return names
+
+    def raise_under_iteration(self, action: str) -> None:
+        if self._under_iteration.is_set():
+            raise RuntimeError(f"Can't {action} while iterating input work items")
+
+    @keyword
+    def for_each_input_work_item(self, keyword, *args, **kwargs):
+        """Run a keyword for each work item in the input queue.
+
+        Returns a list of results.
+        """
+
+        outputs = []
+        self.raise_under_iteration("iterate input work items")
+        self._under_iteration.set()
+
+        try:
+            while True:
+                output = BuiltIn().run_keyword(keyword, *args, **kwargs)
```

Comment:
I guess that this row will fail on pure Python run

Reply:
I added now the possibility of providing a Py function instead of just a RF keyword. Is that what we'd expect?